src/main/java/org/snomed/snowstorm/core/data/services/identifier/LocalSequentialIdentifierSource.java

```diff
@@ -132,10 +135,17 @@ private int findHighestIdentifierSequence(int namespaceId, String partitionId) {
 			} else {
 				highestSctid = component.getId();
 			}
+			String itemIdentifier;
 			if (namespaceId == 0) {
-				highestSequence = Integer.parseInt(highestSctid.substring(0, highestSctid.length() - 3));
+				itemIdentifier = highestSctid.substring(0, highestSctid.length() - 3);
 			} else {
-				highestSequence = Integer.parseInt(highestSctid.substring(0, highestSctid.lastIndexOf(namespaceId + "")));
+				itemIdentifier = highestSctid.substring(0, highestSctid.lastIndexOf(namespaceId + ""));
+			}
+			if (itemIdentifier.length() <= 8) {
+				highestSequence = Integer.parseInt(itemIdentifier);
+			} else {
+				logger.error("{} found with bad SCTID '{}' on branch {}, this will prevent identifier generation.",
+						componentClass.getSimpleName(), highestSctid, component.getPath());
```

Comment:
It is a tricky scenario here. Assigning next sequence number to 100 if the SCTID found is invalid. It depends on the use case I think. Maybe snowstorm fails to start makes sense in some case because not everyone is checking the logs. Or maybe could we log error and switch to local random identifier when this happens?

Reply:
I think this block was just to support unit testing where we had some badly made up data. I am comfortable not allowing ID generation to work correctly if there are bad SCTIDs in the data. 
The sequence number is being set of 100 for the International Edition to support unit testing when there is no data loaded, otherwise I think the generated ID is too short and doesn't match the spec. This would never happen because the International Edition would be loaded first and anyway it's maintained on the Managed Service using the CIS service.

Reply:
As long as this is acceptable then logging error is fine by me.